### PR TITLE
Remove references to legacy visibility system

### PR DESCRIPTION
**Pull request by hultonha**
This is a partial feature removal - it would be great to remove more legacy code but as we all know tugging at that thread can unravel very quickly 🙈 Think of this as a work in progress to gradually remove all need for `CObjectManager` and `CBaseObject`.

This removes the option to use the old object visibility cache (it hasn't been used for sometime already) and removes code that should never be actually called (actually deleting the functions can wait as it quickly begins to spiral, I wanted to keep this change as small as possible).

### CODE REVIEW

File: Code/Editor/Viewport.cpp [hunk 1]
```diff
@@ -398,11 +389,8 @@ void QtViewport::Update()
     m_viewportUi.Update();
 
     m_bAdvancedSelectMode = false;
-    bool bSpaceClick = false;
-    {
-        bSpaceClick = CheckVirtualKey(Qt::Key_Space) & !CheckVirtualKey(Qt::Key_Shift) /*& !CheckVirtualKey(Qt::Key_Control)*/;
-    }
-    if (bSpaceClick && hasFocus())
+
+    if (CheckVirtualKey(Qt::Key_Space) && !CheckVirtualKey(Qt::Key_Shift) && hasFocus())
```
Comment (NicholasVanSickle) [inline Code/Editor/Viewport.cpp:393]: Any idea if this and `QtViewport::HitTest` can go away any time soon? I know I've seen this and considered pulling it out before, I think this is one of the last bits of hardcoded input handling in the viewport.

**Review by NicholasVanSickle — APPROVED** [review]
Change looks great, always feels good to +1 redcode

File: Code/Editor/Viewport.cpp (hunk 1 above)
Comment (hultonha) [reply Code/Editor/Viewport.cpp:393]: That's a good point, it most likely can and I can try that and see what else complains in a follow-up PR 🙂 Thanks!

File: Code/Editor/Objects/ObjectManager.cpp [hunk 2]
```diff
@@ -301,12 +303,13 @@ CBaseObject* CObjectManager::NewObject(CObjectArchive& ar, CBaseObject* pUndoObj
...
                 errorRecord.error = error;
-                errorRecord.description = "Possible duplicate objects being loaded, potential fix is to remove duplicate objects from level files.";
+                errorRecord.description =
+                    "Possible duplicate objects being loaded, potential fix is to remove duplicate objects from level files.";
                 GetIEditor()->GetErrorReport()->ReportError(errorRecord);
             }
 
             return nullptr;
-            //CoCreateGuid( &pObject->m_guid ); // generate uniq GUID for this object.
+            // CoCreateGuid( &pObject->m_guid ); // generate uniq GUID for this object.
```
Comment (AMZN-daimini) [inline Code/Editor/Objects/ObjectManager.cpp:312]: nit: remove commented out code?

File: Code/Editor/Objects/ObjectManager.cpp [hunk 3]
```diff
@@ -323,7 +326,7 @@ CBaseObject* CObjectManager::NewObject(CObjectArchive& ar, CBaseObject* pUndoObj
         return nullptr;
     }
 
-    //pObject->Serialize( ar );
+    // pObject->Serialize( ar );
```
Comment (AMZN-daimini) [inline Code/Editor/Objects/ObjectManager.cpp:329]: nit: remove commented out code?

File: Code/Editor/Objects/ObjectManager.cpp [hunk 4]
```diff
@@ -521,9 +525,9 @@ CBaseObject* CObjectManager::CloneObject(CBaseObject* obj)
 {
     AZ_PROFILE_FUNCTION(Editor);
     assert(obj);
-    //CRuntimeClass *cls = obj->GetRuntimeClass();
-    //CBaseObject *clone = (CBaseObject*)cls->CreateObject();
-    //clone->CloneCopy( obj );
+    // CRuntimeClass *cls = obj->GetRuntimeClass();
+    // CBaseObject *clone = (CBaseObject*)cls->CreateObject();
+    // clone->CloneCopy( obj );
```
Comment (AMZN-daimini) [inline Code/Editor/Objects/ObjectManager.cpp:530]: nit: remove commented out code?

File: Code/Editor/Objects/ObjectManager.cpp [hunk 5]
```diff
@@ -1434,7 +1343,7 @@ void CObjectManager::EndEditParams([[maybe_unused]] int flags)
 {
     m_bSingleSelection = false;
     m_currEditObject = nullptr;
-    //m_bSelectionChanged = false; // don't need to clear for ungroup
+    // m_bSelectionChanged = false; // don't need to clear for ungroup
```
Comment (AMZN-daimini) [inline Code/Editor/Objects/ObjectManager.cpp:1346]: nit: remove commented out code?

**Review by AMZN-daimini — APPROVED** [review]

File: Code/Editor/Objects/ObjectManager.cpp (hunk 5 above)
Comment (hultonha) [reply Code/Editor/Objects/ObjectManager.cpp:1346]: Oh crap I accidentally ran clang-format on that file 🙈 I'll revert those changes and re-upload sorry 😩

**Comment by hultonha** [comment]
@AMZN-daimini Sorry about the formatting changes, that was unintentional 🤦‍♂️ (clang-format is now in my muscle memory like Ctrl-S is 😬) I have reverted them now and the diff is much cleaner

**Comment by AMZN-daimini** [comment]
> @AMZN-daimini Sorry about the formatting changes, that was unintentional 🤦‍♂️ (clang-format is now in my muscle memory like Ctrl-S is 😬) I have reverted them now and the diff is much cleaner

I didn't mind the clang fixes, but yeah this is much easier to read now :D thanks!

**Review by bosnichd — COMMENTED** [review]
Nice, thanks!

**Pull request merged**